controllers/machine_controller.go

```diff
@@ -103,7 +103,7 @@ func (r *MachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manag
 		&source.Kind{Type: &clusterv1.Cluster{}},
 		handler.EnqueueRequestsFromMapFunc(clusterToMachines),
 		// TODO: should this wait for Cluster.Status.InfrastructureReady similar to Infra Machine resources?
-		predicates.ClusterUnpaused(ctrl.LoggerFrom(ctx)),
+		predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue),
```

Comment:
Maybe I'm wrong but is that really the same? ClusterUnpaused seems to check .spec.Paused while ResourceNotPaused checks the annotation.

Do we also set the annotation on the cluster resource?

Reply:
I'm not sure. I'm changing it to `ClusterUnpaused` just in case, please take a look.